core/src/main/java/org/elasticsearch/rest/action/support/RestStatusToXContentListener.java

```diff
@@ -37,8 +37,10 @@
      * Build an instance that doesn't support responses with the status {@code 201 CREATED}.
      */
     public RestStatusToXContentListener(RestChannel channel) {
-        // TODO switch this to throwing an exception?
-        this(channel, r -> null);
+        this(channel, r -> {
+            assert false: "Returned a 201 CREATED but not set up to support a Location header";
```

Comment:
sorry I just saw your comment [here](https://github.com/elastic/elasticsearch/pull/19509#issuecomment-235745806). I guess this will be a sanity check to verify that we didn't miss anything.